### PR TITLE
Allowing choosing pointers to Scalars in the GUI

### PR DESCRIPTION
**Pull request by pconesa**
Also, validation errors at runtime do not print the trace.

### CODE REVIEW

File: pyworkflow/gui/form.py [hunk 1]
```diff
@@ -1855,6 +2114,13 @@ def getWidgetValue(self, protVar, param):
             isinstance(param, params.MultiPointerParam) or
             isinstance(param, params.RelationParam)):
             widgetValue = protVar
+        # For Scalar params that allowPointers
+        elif param.allowPointers:
+            if protVar.hasPointer():
+                # NOTE: to check with JM, we should avoid accessing _pointer
+                widgetValue = protVar._pointer
```
Comment (delarosatrevin) [inline pyworkflow/gui/form.py]: I added the fuction, so now you could use: protVar.getPointer()

**Comment by dmaluenda** [comment]
I'm getting the following error when I chose and select a setOfSomething for any input of any protocol in the paramWidget (browser).

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/home/malu/scipionRELEASE/software/lib/python2.7/lib-tk/Tkinter.py", line 1541, in __call__
    return self.func(*args)
  File "/home/malu/scipionRELEASE/pyworkflow/gui/form.py", line 1374, in _onVarChanged
    self.callback(self.paramName)
  File "/home/malu/scipionRELEASE/pyworkflow/gui/form.py", line 2248, in _checkChanges
    self.setParamFromVar(paramName)
  File "/home/malu/scipionRELEASE/pyworkflow/gui/form.py", line 2329, in setParamFromVar
    param.copy(widget)
  File "/home/malu/scipionRELEASE/pyworkflow/object.py", line 431, in copy
    self._copy(other, copyDict, copyId, ignoreAttrs=ignoreAttrs)
  File "/home/malu/scipionRELEASE/pyworkflow/object.py", line 457, in _copy
    self._objId = other._objId
AttributeError: ParamWidget instance has no attribute '_objId'
```

When I checkout devel-pluginization back, this doesn't happens. When I return to  `pc-scalar-pointe` the error is gotten.

**Comment by dmaluenda** [comment]
I've just realized that this error is not happening on the Scipion protocols, only in plugin protocols.

File: pyworkflow/gui/form.py (hunk 1 above)
Comment (pconesa) [reply pyworkflow/gui/form.py]: thanks, I'll change this to use the getter.

**Comment by pconesa** [comment]
ummm, I'll double check that.

**Comment by pconesa** [comment]
Got it, I've introduced this error while renaming variables. grrr!

**Comment by pconesa** [comment]
[x] Pointer params (allowsNull and not): tested on extract particles (ctf for nullable pointer)
[x] PointerList params (adding/removing to the selection): tested on the summary monitor
[x] Scalar values are still possible to enter by typing, even if the 'allowsPointer' is set to True.
[x] Scalar values are properly used, both if set directly or by pointer.

File: pyworkflow/gui/form.py (hunk 1 above)
Comment (pconesa) [reply pyworkflow/gui/form.py]: Done too!

**Comment by dmaluenda** [comment]
I've tested and it work pretty fine in the following cases (except for the laast one):
[x] `allowPointers=True` in combination with `allowsNull=True` in the same parameter and giving an object.
[x] `allowPointers=True` in combination with `allowsNull=True` in the same parameter and giving a value.
[x] `allowPointers=True` in combination with `allowsNull=True` in the same parameter and giving nothing.
[x]  `allowPointers=True`, giving an object and then running the wizard. The wizard takes the object value (as expected), the same for the direct value (former way).
[ ]  `allowPointers=True` in combination with `allowsNull=True` in the same parameter where I putted an object and then it is deleted (resulting an empty field). **The protocol takes the last object**...

For the last case (which is failing since we expect a None value), I think it would be great if the trash icon is shown. I've deleted the object by putting the cursor there and pressing the 'del' key from the keyboard. I think that in this way the value remains loaded in the protocol class and it's not updated.

Very great job!!!

**Comment by dmaluenda** [comment]
Just a minor, but useful I guess, I think we can standardize the option name. We have the ''allow**s**Null" option, so I will suggest the "allow**s**Pointer~s~".

**Comment by pconesa** [comment]
Thank you @dmaluenda  for the detailed report. I'll rename the parameter adding an "s".. and check the behaviour  in combination with allowsNull.

**Comment by dmaluenda** [comment]
@pconesa also have a look on how to delete an object if, finally, the user want to pass nothing.

**Comment by pconesa** [comment]
In theory...emptying the box should empty the params

**Comment by pconesa** [comment]
I rather avoid the "bin" for this case.

**Comment by dmaluenda** [comment]
In theory, but as I pointed out above, when you select an object using the lens and, after, that object is deleted; the protocol is still taking that deleted object.

 > [ ] allowPointers=True in combination with allowsNull=True in the same parameter where I putted an object and then it is deleted (resulting an empty field). The protocol takes the last object ...

**Comment by pconesa** [comment]
I'll look at that and make it work.

**Comment by pconesa** [comment]
All addressed now, thank you both! (@delarosatrevin and @dmaluenda )

**Pull request merged**